frame/clearing-house/src/lib.rs

```diff
@@ -2241,10 +2285,18 @@ pub mod pallet {
 	// Oracle update helpers
 	impl<T: Config> Pallet<T> {
 		fn update_oracle_twap(market: &mut Market<T>) -> Result<(), DispatchError> {
-			let mut oracle = Self::clip_around_twap(
-				&Self::oracle_price(market.asset_id)?,
-				&market.last_oracle_twap,
-			)?;
+			let mut oracle = Self::oracle_price(market.asset_id)?;
+			ensure!(oracle.is_positive(), Error::<T>::InvalidOracleReading);
```

Comment:
In line 508 the definition of `InvalidOracleReading` is: *Raised when the price returned by the Oracle is zero.*,
However, at least theoretically there could be a case where the oracle value is negative since `Decimal` is signed, right?

In that case wouldn't be better to define this error alluding to the price being strictly positive?

Reply:
Looking at the `Oracle` trait, which is where the Clearing House gets its prices from, `get_price` is returned as a `Balance`, thus it is nonnegative.

But at a higher level, all prices should be positive indeed. The error of the price being zero is just one instance of the oracle reading being invalid. So I agree with your suggestion

Reply:
Specially because we may switch to an Oracle which uses signed quantities in the futures and still use the same error